### ISSUE TITLE
General exception occured when attempting to insert record

### ISSUE THREAD

**Issue by marsrobertson** [rkalis/truffle-plugin-verify#71]
I'm trying to deploy this: https://github.com/basexisland/assurance-contract

Need to change `secrets.json.sample` into `secrets.json` and provide:
* private key
* infura
* etherscan

```
m@hck network-state-genesis % truffle migrate --network rinkeby

Compiling your contracts...
===========================
✔ Fetching solc version list from solc-bin. Attempt #1
> Compiling ./contracts/DummyTestingWBTC.sol
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/NetworkStateGenesis.sol
> Compiling ./node_modules/@openzeppelin/contracts/access/AccessControl.sol
> Compiling ./node_modules/@openzeppelin/contracts/access/Ownable.sol
> Compiling ./node_modules/@openzeppelin/contracts/introspection/ERC165.sol
> Compiling ./node_modules/@openzeppelin/contracts/introspection/IERC165.sol
> Compiling ./node_modules/@openzeppelin/contracts/math/SafeMath.sol
> Compiling ./node_modules/@openzeppelin/contracts/presets/ERC20PresetMinterPauser.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC20/ERC20.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC20/ERC20Pausable.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC20/IERC20.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC721/ERC721.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC721/IERC721.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC721/IERC721Enumerable.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC721/IERC721Metadata.sol
> Compiling ./node_modules/@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol
> Compiling ./node_modules/@openzeppelin/contracts/utils/Address.sol
> Compiling ./node_modules/@openzeppelin/contracts/utils/Context.sol
> Compiling ./node_modules/@openzeppelin/contracts/utils/EnumerableMap.sol
> Compiling ./node_modules/@openzeppelin/contracts/utils/EnumerableSet.sol
> Compiling ./node_modules/@openzeppelin/contracts/utils/Pausable.sol
> Compiling ./node_modules/@openzeppelin/contracts/utils/Strings.sol
✔ Fetching solc version list from solc-bin. Attempt #1
> Artifacts written to /Users/m/Code/network-state-genesis/build/contracts
> Compiled successfully using:
   - solc: 0.6.12+commit.27d51765.Emscripten.clang



Starting migrations...
======================
> Network name:    'rinkeby'
> Network id:      4
> Block gas limit: 10000000 (0x989680)


1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------
   > transaction hash:    0x6907c8c3c967c2b50ddfbf7f652c86cc6157f9ad40e83eefce520d6f9e8628b5
   > Blocks: 3            Seconds: 40
   > contract address:    0x06789e99eC694C33059F5F8411D2484db5f8da78
   > block number:        8384723
   > block timestamp:     1618004548
   > account:             0x901CF8126c5F71f904086491FA71Ec47B5E5505d
   > balance:             0.511414565
   > gas used:            130486 (0x1fdb6)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.00260972 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.00260972 ETH


2_deploy_contracts.js
=====================

   Deploying 'NetworkStateGenesis'
   -------------------------------
   > transaction hash:    0xb6ad9b7eb32285188bdfc903864ececc09b66eb307e35759dda13cede7253175
   > Blocks: 1            Seconds: 24
   > contract address:    0x00F2e09676D112DaB5A0368cdfdCC5b9b4B00806
   > block number:        8384728
   > block timestamp:     1618004623
   > account:             0x901CF8126c5F71f904086491FA71Ec47B5E5505d
   > balance:             0.458360965
   > gas used:            2607014 (0x27c7a6)
   > gas price:           20 gwei
   > value sent:          0 ETH
   > total cost:          0.05214028 ETH


   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:          0.05214028 ETH


Summary
=======
> Total deployments:   2
> Final cost:          0.05475 ETH


m@hck network-state-genesis % truffle run verify NetworkStateGenesis --network rinkeby
Verifying NetworkStateGenesis
General exception occured when attempting to insert record
Failed to verify 1 contract(s): NetworkStateGenesis
```

Tried resolve using: https://github.com/rkalis/truffle-plugin-verify/issues/52#issuecomment-770924240

```
m@hck network-state-genesis % npm install truffle-plugin-verify@legacy
npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410
npm WARN @babel/helper-compilation-targets@7.13.13 requires a peer of @babel/core@^7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @babel/helper-define-polyfill-provider@0.2.0 requires a peer of @babel/core@^7.4.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN @babel/plugin-transform-runtime@7.13.15 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN babel-plugin-polyfill-corejs2@0.2.0 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN babel-plugin-polyfill-corejs3@0.2.0 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN babel-plugin-polyfill-regenerator@0.2.0 requires a peer of @babel/core@^7.0.0-0 but none is installed. You must install peer dependencies yourself.

+ truffle-plugin-verify@0.4.1
added 24 packages from 13 contributors, updated 5 packages and audited 571 packages in 8.019s

57 packages are looking for funding
  run `npm fund` for details

found 6 vulnerabilities (5 low, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
m@hck network-state-genesis % truffle run verify NetworkStateGenesis --network rinkeby
Verifying NetworkStateGenesis
Found duplicate SPDX-License-Identifiers in the Solidity code, please provide the correct license with --license <license identifier>
Failed to verify 1 contract(s): NetworkStateGenesis
m@hck network-state-genesis % truffle run verify NetworkStateGenesis --network rinkeby --license MIT
Verifying NetworkStateGenesis
General exception occured when attempting to insert record
Failed to verify 1 contract(s): NetworkStateGenesis
```

# Still doesn't work... Any ideas?

**Comment by marsrobertson** [comment]
`await deployer.deploy(NetworkStateGenesis, "Network State Genesis", "🔮", multisig, WBTCaddress, cutoffTimestamp);`

Maybe emoji problem?

**Comment by marsrobertson** [comment]
It was emoji issue after all!

_(there was another emoji in the code)_

**Comment by felixakiragreen** [comment]
I think I am running into a similar issue where like an idiot I included some special characters in the comments. Were you able to find a workaround for getting your contract verified?

**Comment by rkalis** [comment]
Closing this as a duplicate of #56
